src/accordion.ts

```diff
@@ -0,0 +1,185 @@
+            ) {
+                this._openItem(this.items[0], 0);
+            }
+        }
+    }
+
+    private handleSlotChange(e: Event) {
+        const slot = e.target as HTMLSlotElement;
+        const newItems = slot.assignedElements({
+            flatten: true,
+        }) as HTMLElement[];
+
+        newItems.forEach((item, index) => {
+            if (this.processedItems.has(item)) {
+                return;
+            }
+
+            this.processedItems.add(item);
+            const toggleElement = item.querySelector(this.togglesel);
+            if (toggleElement) {
+                toggleElement.addEventListener("click", () =>
+                    this.toggle(index)
+                );
+            }
+            if (item.classList.contains(this.activecls)) {
+                this._openItem(item, index, false);
+            } else {
+                const body = item.querySelector(this.bodysel) as HTMLElement;
+                if (body) {
+                    body.style.display = "none";
+                }
+            }
+        });
+    }
+
+    private _openItem(item: Element, index: number, animate: boolean = true) {
+        if (this.nomultiple) {
+            this.openItems.forEach((openItem) => {
+                if (openItem !== item) {
+                    this._closeItem(openItem, index);
+                }
+            });
+        }
+        if (this.nomultiple && this.nocloseall) {
+            this.openItems.forEach((openItem) => {
+                if (openItem !== item) {
+                    this._closeItem(openItem, index);
+                }
+            });
+        }
+
+        item.classList.add(this.activecls);
+        const body = item.querySelector(this.bodysel) as HTMLElement;
+        if (body) {
+            body.style.display = "block";
+
+            if (animate && this.animationDur !== 0) {
+                let height = body.scrollHeight;
+                body.style.overflow = "hidden";
+                body.style.height = "0px";
+                const fullHeight = height + "px";
+                body.style.transitionProperty = "height";
+                body.style.transitionDuration = `${this.animationDur}ms`;
+                body.style.transitionTimingFunction = "ease";
+
+                requestAnimationFrame(() => {
+                    body.style.height = fullHeight;
+                });
+            }
+        }
+        if (!this.openItems.includes(item)) {
+            this.openItems.push(item);
+        }
+        this.dispatchEvent(
+            new CustomEvent("itemopened", { detail: { index } })
+        );
+    }
+
+    private _closeItem(item: Element, index: number, animate: boolean = true) {
+        const isOpen = item.classList.contains(this.activecls);
+        if (isOpen && this.nocloseall && this.nomultiple) {
+            item.classList.remove(this.activecls);
+        }
+        if (isOpen && this.nocloseall && this.openItems.length <= 1) {
+            return;
+        }
```

Comment:
This is not correct. Try to reason why? Add a active class name definition in the dev/accordion.html file and enable "no multipe"  and try closing the last opened item and see if the behaviour is correct. 
Also, this can be written in a better way.